Some QC report tweaks (#364)
AlexsLemonade/scpca-nf: templates/qc_report/cite_qc.rmd

```{r}
antibody_tags <- as.data.frame(rowData(adt_exp)) |>
  tibble::rownames_to_column("Antibody") |>
  # ensure `target` is the first level of target_type
  mutate(target_type = forcats::fct_relevel(target_type, "target")) |>
  # group by target_type before arranging
  group_by(target_type) |>
  # use .by_group = TRUE to arrange by target type first, then by mean

>   arrange(desc(mean), .by_group = TRUE) |>

Author: Am I crazy or is this a new-ish feature then? I feel like every time I've tried to make `arrange` work with two variables it has failed me...

>   arrange(desc(mean), .by_group = TRUE) |>

Author: honestly i'm going with crazy...

```{r}
antibody_tags <- as.data.frame(rowData(adt_exp)) |>
  tibble::rownames_to_column("Antibody") |>
  # ensure `target` is the first level of target_type
  mutate(target_type = forcats::fct_relevel(target_type, "target")) |>

>   arrange(target_type, desc(mean)) |> 

Author: > When we have a lot of ADTs, this table gets kind of unreasonably long. 

_yes, yes it does._

I'll open an issue for circling back to this.

```{r fig.alt="Cell filtering based on both ADT and RNA counts", results='asis'}
if (has_processed) { 
  
  glue::glue('
    The plot below displays an overall view of cell filtering based on both ADT and RNA counts. Cells are labeled as one of the following:
  
    - "Keep": This cell is retained based on both RNA and ADT counts.
    - "Filter (RNA only)": This cell is filtered based on only RNA counts.

>     - "Filter (ADT only)" This cell is filtered based on only ADT counts.

Reviewer: Missing one colon
```suggestion
    - "Filter (ADT only)": This cell is filtered based on only ADT counts.
```

if (has_processed) { 
  

>   # DO NOT TAB TEXT IN, for formatting reasons

Reviewer: I'm guessing you didn't add all of this arbitrarily, but it is not in keeping with my understanding of the functions involved, so I want to push it a bit in case we can get rid of some extra formatting.

My main thought is that`glue` ignores leading spaces, unless you tell it to keep them. So I would expect the following to work just fine, based on my experience with `glue` and some quick testing in a separate notebook:

```
  glue::glue('
    The plot below displays an overall view of cell filtering based on both ADT and RNA counts. Cells are labeled as one of the following:
  
    - "Keep": This cell is retained based on both RNA and ADT counts.
    - "Filter (RNA only)": This cell is filtered based on only RNA counts.
    - "Filter (ADT only)" This cell is filtered based on only ADT counts.
    - "Filter (RNA & ADT)": This cell is filtered based on both RNA and ADT counts.
  
  ')  |> print()
```

This is following the rules described in https://glue.tidyverse.org/reference/trim.html. 
I also removed all of the explicit `\n`, because I have never before needed those with `glue`. 
I might imagine you could need a final `\n`, but making that literal (as above) should be fine too.

Author: :100: thank you, i got so turned around in my formatting journey it seems...

  glue::glue('
The plot below displays an overall view of cell filtering based on both ADT and RNA counts. Cells are labeled as one of the following:\n\n

- "Keep": This cell is retained based on both RNA and ADT counts.\n
- "Filter (RNA only)": This cell is filtered based on only RNA counts.\n
- "Filter (ADT only)" This cell is filtered based on only ADT counts.\n\n

> - "Filter (RNA & ADT)": This cell is filtered based on both RNA and ADT counts.\n

Reviewer: Just noticing all the `\n` here, which I would not have expected to be needed.

```

```{r fig.alt="Density plot showing normalized expression of highly variable ADTs", warning = FALSE, results='asis'}
if (has_processed) {
  
  glue::glue(

> "The plot below displays normalized expression of these four ADTs, with one ADT shown per panel.\n\n"

Reviewer: I feel like one trailing newline ought to be enough?
```suggestion
    "The plot below displays normalized expression of these four ADTs, with one ADT shown per panel.\n"
```

} 
```

```{r fig.alt="UMAP calculated from RNA expression but colored by normalized expression of highly variable ADTs", fig.height = 6, results='asis'}
if (has_processed) {
    

>   # 2 \n\n here avoids text wrapping the plot in wide screens
>   glue::glue(
>   "The plot below displays UMAP embeddings calculated from RNA expression, where each cell is colored by the expression level of the given ADT.\n\n"
>   ) |> print() 

Reviewer: An alternative version here. Mostly me being opposed to `\n`, but not important.
```suggestion
  # Extra blank line here keeps the plot in a separate paragraph
  glue::glue(
    "The plot below displays UMAP embeddings calculated from RNA expression, where each cell is colored by the expression level of the given ADT.
    
    "
  ) |> print() 
```

Author: Fine with me, I'll test it!

Author: Noting this does prevent the ugly wrapping!

} 
```

```{r fig.alt="UMAP calculated from RNA expression but colored by normalized expression of highly variable ADTs", fig.height = 6, results='asis'}
if (has_processed) {
    
  glue::glue(

> "The plot below displays UMAP embeddings calculated from RNA expression, where each cell is colored by the expression level of the given ADT.\n\n"

Reviewer: ```suggestion
    "The plot below displays UMAP embeddings calculated from RNA expression, where each cell is colored by the expression level of the given ADT.\n\n"
```

Author: done separately.
